distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Upload_Table.enso

```
@@ -229,20 +225,58 @@ common_update_table source_table connection table_name update_action key_columns
+                    insert_statement = connection.dialect.generate_sql <|
+                        Query.Insert_From_Select target_table.name source_table.column_names source_table.to_select_query
+                    Panic.rethrow <| connection.execute_update insert_statement
+            Update_Action.Update -> throw_not_implemented
+            Update_Action.Update_Or_Insert -> throw_not_implemented
+            Update_Action.Align_Records -> throw_not_implemented
+        upload_status.if_not_error target_table
+
+## PRIVATE
+check_update_arguments source_table target_table key_columns update_action error_on_missing_columns on_problems ~action =
     source_columns = Set.from_vector source_table.column_names
     target_columns = Set.from_vector target_table.column_names
     extra_columns = source_columns.difference target_columns
-    if extra_columns.not_empty then Error.throw (Unmatched_Columns.Error extra_columns) else
+    if extra_columns.not_empty then Error.throw (Unmatched_Columns.Error extra_columns.to_vector) else
         missing_columns = target_columns.difference source_columns
-        if missing_columns.not_empty && error_on_missing_columns then Error.throw (Missing_Input_Columns.Error missing_columns "the source table") else
-            # TODO [RW] to be finished in follow up PR
-            _ = [update_action, key_columns]
-            Error.throw (Illegal_State.Error "TODO: Not implemented yet.")
+        if missing_columns.not_empty && error_on_missing_columns then Error.throw (Missing_Input_Columns.Error missing_columns.to_vector "the source table") else
+            key_set = Set.from_vector key_columns
+            missing_source_key_columns = key_set.difference source_columns
+            missing_target_key_columns = key_set.difference target_columns
+            # TODO Shall we add some 'location' to `Missing_Input_Columns` to distinguish when a missing column comes from table structure VS `key_columns`?
```

I don't think we need to - we will raise what is missing, and the user will have to fix.